janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -143,45 +153,63 @@ public void clusterHealthRequest(String timeout) throws IOException {
 
     @Override
     public boolean indexExists(String indexName) throws IOException {
-        boolean exists = false;
-        try {
-            delegate.performRequest(REQUEST_TYPE_GET, REQUEST_SEPARATOR + indexName);
-            exists = true;
-        } catch (final IOException e) {
-            if (!e.getMessage().contains("404 Not Found")) {
-                throw e;
-            }
-        }
-        return exists;
+        final Response response = delegate.performRequest(new Request(REQUEST_TYPE_HEAD, REQUEST_SEPARATOR + indexName));
+        return response.getStatusLine().getStatusCode() == 200;
     }
 
     @Override
     public boolean isIndex(String indexName) {
-        boolean exists = false;
         try {
-            final Response response = delegate.performRequest(REQUEST_TYPE_GET, REQUEST_SEPARATOR + indexName);
+            final Response response = delegate.performRequest(new Request(REQUEST_TYPE_GET, REQUEST_SEPARATOR + indexName));
             try (final InputStream inputStream = response.getEntity().getContent()) {
-                exists = mapper.readValue(inputStream, Map.class).containsKey(indexName);
+                return mapper.readValue(inputStream, Map.class).containsKey(indexName);
             }
         } catch (final IOException ignored) {
         }
-        return exists;
+        return false;
     }
 
     @Override
     public boolean isAlias(String aliasName)  {
-        boolean exists = false;
         try {
-            delegate.performRequest(REQUEST_TYPE_GET, REQUEST_SEPARATOR + "_alias" + REQUEST_SEPARATOR + aliasName);
-            exists = true;
+            delegate.performRequest(new Request(REQUEST_TYPE_GET, REQUEST_SEPARATOR + "_alias" + REQUEST_SEPARATOR + aliasName));
+            return true;
         } catch (final IOException ignored) {
         }
-        return exists;
+        return false;
     }
 
     @Override
     public void createIndex(String indexName, Map<String,Object> settings) throws IOException {
-        performRequest(REQUEST_TYPE_PUT, REQUEST_SEPARATOR + indexName, mapWriter.writeValueAsBytes(settings));
+
+        Request request = new Request(REQUEST_TYPE_PUT, REQUEST_SEPARATOR + indexName);
+
+        if(ElasticMajorVersion.SEVEN.equals(majorVersion)){
+            // TODO: Temporary solution to support mapping types in ES7
+            request.addParameter(INCLUDE_TYPE_NAME_PARAMETER, "true");
```

Comment:
What do you think of temporary? Is this problem in the future?

Reply:
The parameter `include_type_name` is set to `false` in ES 7 which indicates that we should not use mapping types. This parameter will be removed in ES 8 and mapping types won't be supported anymore. 
Maybe the word `Temporary` isn't clear here. Right now we keep mapping types for ES 7 in this PR and that is why it will be easy for users to upgrade to ES 7 without need to reindex data. But it will require reindex for ES 8 than. We may also decide to disable mapping types for ES 7 in 0.5.0 version of JanusGraph which will give users easy migration from ES 7 to ES 8 but will require reindex from ES 6 to ES 7. I will work on mapping types removal in the future PRs. Maybe I will add one more parameter to the JanusGraph which enables or disables mappings for ES 7. It will give users with the choice to either reindex the data now to have easy migration in the future or to keep mapping for ES 7 and make reindexing in the future if they decide to migrate to ES 8.

Reply:
I just said `Temporary` because we may change the behavior when mapping types removal PR is prepared

Reply:
This sounds reasonable for me.